tests/shadow/test_shadow_entropies.py

```diff
@@ -0,0 +1,114 @@
+import pennylane.numpy as np
+from pennylane.shadows import ClassicalShadow
+
+np.random.seed(777)
+
+
+def max_entangled_circuit(wires, shots=10000, interface="autograd"):
+    """maximally entangled state preparation circuit"""
+    dev = qml.device("default.qubit", wires=wires, shots=shots)
+
+    @qml.qnode(dev, interface=interface)
+    def circuit():
+        qml.Hadamard(wires=0)
+        for i in range(1, wires):
+            qml.CNOT(wires=[0, i])
+        return qml.classical_shadow(wires=range(wires))
+
+    return circuit
+
+
+class TestShadowEntropies:
+    """Tests for entropies in ClassicalShadow class"""
+
+    @pytest.mark.parametrize("n_wires", [2, 4])
+    @pytest.mark.parametrize("base", [np.exp(1), 2])
+    def test_constant_distribution(self, n_wires, base):
+        """Test for state with constant eigenvalues of reduced state that all entropies are the same"""
+
+        bits, recipes = max_entangled_circuit(wires=n_wires)()
+        shadow = ClassicalShadow(bits, recipes)
+
+        entropies = [
+            shadow.entropy(wires=[0], alpha=alpha, atol=1e-2, base=base) for alpha in [1, 2, 3]
+        ]
+        assert np.allclose(entropies, entropies[0], atol=1e-2)
+
+    def test_non_constant_distribution(
+        self,
+    ):
+        """Test entropies match roughly with exact solution for a non-constant distribution"""
+        n_wires = 4
+        # exact solution
+        dev = qml.device("default.qubit", wires=range(n_wires), shots=100000)
+
+        @qml.qnode(dev)
+        def qnode_exact(x):
+            for i in range(n_wires):
+                qml.RY(x[i], wires=i)
+
+            for i in range(n_wires - 1):
+                qml.CNOT((i, i + 1))
+
+            return qml.state()
+
+        # classical shadow qnode
+        @qml.qnode(dev)
+        def qnode(x):
+            for i in range(n_wires):
+                qml.RY(x[i], wires=i)
+
+            for i in range(n_wires - 1):
+                qml.CNOT((i, i + 1))
+
+            return qml.classical_shadow(wires=dev.wires)
+
+        x = np.arange(n_wires, requires_grad=True)
+
+        bitstrings, recipes = qnode(x)
+        shadow = ClassicalShadow(bitstrings, recipes)
+
+        # Check for the correct entropies for all possible 2-site reduced density matrix (rdm)
+        for rdm_wires in [[0, 1], [0, 2], [0, 3], [1, 2], [1, 3], [2, 3]]:
+            # this is intentionally not done in a parametrize loop because this would re-execute the quantum function
+
+            # exact solution
+            rdm = qml.qinfo.reduced_dm(qnode_exact, wires=rdm_wires)(x)
+            evs = qml.math.eigvalsh(rdm)
+            print(np.round(evs, 3))
+            evs = evs[np.where(evs > 0)]
+
+            exact_2 = -np.log(np.trace(rdm @ rdm))
+
+            alpha = 1.5
+            exact_alpha = qml.math.log(qml.math.sum(evs**alpha)) / (1 - alpha)
+
+            exact_vn = qml.math.entr(evs)
+            exact = [exact_vn, exact_alpha, exact_2]
```

Comment:
I think it'll be better to have a different expression for what the expected value should be, other than just repeating the calculation from the class. The entropy tests in the `qinfo` module are good examples.

Reply:
Thanks for pointing that out! I created a new test with the analytic expressions from the `qinfo` module. I would still keep this since it is testing whether the shadow approximation works or not (so less about the implementation)